core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/adaptive/AdaptiveCodec.java

```diff
@@ -260,4 +265,65 @@ protected String debugInfo() {
     return this.toString();
   }
 
+  public static VectorUtil checkAndUpdateToChildVector(ColumnVectorInfo vectorInfo, int pageSize,
+      CarbonColumnVector vector, DataType vectorDataType) {
+    VectorUtil vectorUtil = new VectorUtil(pageSize, vector, vectorDataType);
+    Stack<CarbonColumnVector> vectorStack = vectorInfo.getVectorStack();
+    // check and update to child vector info
+    if (vectorStack != null && vectorStack.peek() != null && vectorDataType.isComplexType()) {
```

Comment:
use Objects.nonNull for not null case and for null use Objects.isnull

Reply:
Objects.isnull and Objects.nonNull is internally just doing the same. 
For just "if check" it is not much useful (We have not used in our code for if checks also anywhere) 
**I will use it in the future if I use streams.**

![Screenshot from 2020-08-27 17-43-51](https://user-images.githubusercontent.com/5889404/91441253-7d649d00-e88d-11ea-8d30-aa73af1cd7d6.png)

Reply:
**Also initialized the stack always, checking for empty now. no null check is present**